zhaquirks/legrand/remote_switch.py

```diff
@@ -0,0 +1,218 @@
+
+MANUFACTURER_SPECIFIC_CLUSTER_ID = 0xFC01  # decimal = 64513
+
+VOLTAGE_REPORTED = "voltage_reported"
+
+
+class LegrandRemoteSwitchCluster(CustomCluster, ManufacturerSpecificCluster):
+    """LegrandRemoteSwitchCluster."""
+
+    cluster_id = MANUFACTURER_SPECIFIC_CLUSTER_ID
+    name = "LegrandRemoteSwitchCluster"
+    ep_attribute = "legrand_cluster"
+
+
+class LegrandRemoteSwitchLevelControl(CustomCluster, LevelControl):
+    """LegrandRemoteSwitchLevelControl."""
+
+    def bind(self):
+        """Disable binding to SwitchLevelControl cluster of the coordinator."""
+
+        pass
+
+
+class LegrandRemoteSwitchOnOff(CustomCluster, OnOff):
+    """Legrand Wireless Remote Switch On/Off custom cluster."""
+
+    cluster_id = 0x0006
+    name = "OnOff"
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
+
+    def handle_cluster_request(
+        self,
+        hdr: foundation.ZCLHeader,
+        args: List[Any],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ):
+        """handle_cluster_request."""
+
+        if hdr.command_id in (0x0000, 0x0001, 0x0002):
+            # Send default response
+            if not hdr.frame_control.disable_default_response:
+                self.send_default_rsp(hdr, status=foundation.Status.SUCCESS)
+
+            self.endpoint.device.voltage_bus.listener_event(VOLTAGE_REPORTED)
+        else:
+            return super().handle_cluster_request(
+                hdr, args, dst_addressing=dst_addressing
+            )
+
+
+class LegrandRemoteSwitchPowerConfiguration(PowerConfigurationCluster):
+    """Legrand Wireless Remote Switch On/Off custom cluster."""
+
+    cluster_id = 0x0001
+    name = "PowerConfiguration"
+
+    _CONSTANT_ATTRIBUTES = {
+        0x0031: 10,  # battery_size = CR2032
+        0x0033: 1,  # battery_quantity
+        0x0034: 30,  # battery_rated_voltage = 3000 mV
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.voltage_bus.add_listener(self)
+
+    async def _configure_reporting(self, *args, **kwargs):  # pylint: disable=W0221
+        """Prevent remote configure reporting."""
+
+        return (foundation.ConfigureReportingResponse.deserialize(b"\x00")[0],)
+
+    def voltage_reported(self):
+        """Voltage reported."""
+
+        self.create_catching_task(
+            super().read_attributes(
+                [
+                    self.BATTERY_VOLTAGE_ATTR,
+                ],
+                manufacturer=0x1021,
+            )
+        )
+
+
+class LegrandRemoteSwitchPollControl(CustomCluster, PollControl):
+    """Legrand Wireless Remote Switch PollControl cluster.
+
+    Read PowerConfiguration attributes during check-in.
+    """
+
+    cluster_id = 0x0020
```

Comment:
Cluster id and name will come from parent

Reply:
Fixed